qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -289,3 +302,86 @@ def default_values(self) -> List[DefaultCalValue]:
+
+    __default_values__ = {
+        "duration": 640,
+        "width": 384,
+        "σ": 64,
+        "amp": 0.5,
+        "amp_trg": 0.0,
+    }
+
+    def __init__(
+        self,
+        pulse_on_target: bool = True,
+        basis_gates: Optional[List[str]] = None,
+        default_values: Optional[Dict] = None,
+    ):
+        """Setup the schedules.
+
+        Args:
+            pulse_on_target: A boolean to indicate if a rotary/cancellation tones are
+                used in the CR gate. If this value is set to False then no pulse is played
+                on the target qubit during the cross-resonance pulse.
+            basis_gates: The basis gates to generate.
+            default_values: Default values for the parameters this dictionary can contain
+                the following keys: "duration", "amp", "amp_trg", "width", and "σ".
+        """
+        self._rotary = pulse_on_target
+        super().__init__(basis_gates, default_values)
+
+    @property
+    def __supported_gates__(self) -> Dict[str, int]:
+        """The gates that this library supports and the number of qubits they apply to."""
+        return {"ecr": 2, "cr45p": 2, "cr45m": 2}
+
+    def _build_schedules(self, basis_gates: Set[str]) -> Dict[str, ScheduleBlock]:
+        """Build the schedules of the library."""
+
+        control = pulse.DriveChannel(Parameter("ch0"))
+        target = pulse.DriveChannel(Parameter("ch1"))
+        cr_chan = pulse.ControlChannel(Parameter("ch0.1"))
+
+        cr_amp = Parameter("amp")
+        rot_amp = Parameter("amp_trg")
+        cr_sigma = Parameter("σ")
+        cr_duration = Parameter("duration")
+        cr_width = Parameter("width")
+
+        cr45p = pulse.GaussianSquare(
+            duration=cr_duration, amp=cr_amp, sigma=cr_sigma, width=cr_width, name="cr45p"
+        )
+
+        cr45m = pulse.GaussianSquare(
+            duration=cr_duration, amp=-cr_amp, sigma=cr_sigma, width=cr_width, name="cr45m"
+        )
+
+        cr45p_target = pulse.GaussianSquare(
+            duration=cr_duration, amp=rot_amp, sigma=cr_sigma, width=cr_width, name="cr45p_target"
+        )
+
+        cr45m_target = pulse.GaussianSquare(
+            duration=cr_duration, amp=-rot_amp, sigma=cr_sigma, width=cr_width, name="cr45m_target"
+        )
+
+        with pulse.build(name="cr45p") as cr45p_sched:
+            pulse.play(cr45p, cr_chan)
+            if self._rotary:
+                pulse.play(cr45p_target, target)
+
+        with pulse.build(name="cr45m") as cr45m_sched:
+            pulse.play(cr45m, cr_chan)
+            if self._rotary:
+                pulse.play(cr45m_target, target)
+
+        ecr_sched = pulse.ScheduleBlock(name="ecr", alignment_context=AlignSequential())
+        ecr_sched.append(cr45p_sched)
+        ecr_sched.append(CalledScheduleByName("x", control))
+        ecr_sched.append(cr45m_sched)
+        ecr_sched.append(CalledScheduleByName("x", control))
```

Comment:
@nkanazawa1989 I tried out your suggestion and it didn't work because of this:

https://github.com/Qiskit/qiskit-terra/blob/dde6126e8d8ad550801e9a29adc21c83faf17748/qiskit/pulse/builder.py#L739-L741

```python
        # ignore empty context
        if len(context_block) > 0:
            self._context_stack[-1].append(context_block)
```

Since `CalledScheduleByName` has length 0, it does not get appended. I wonder if length 0 could cause other problems? We could make the length 1 ...or check the terra code and consider removing the length 0 checks?

Reply:
I guess we were just bit lazy to import all builder functions to the top module. I'm fine with importing all of them. 

> Since CalledScheduleByName has length 0, it does not get appended.

Hmmm this looks like a problem. I don't quite remember why it is here. Perhaps something relevant to block to schedule conversion, or simply just for data efficiency. Of course we can override `__len__` of schedule placeholder (I prefer this solution).